### ISSUE TITLE
Compare error

### ISSUE THREAD

**Issue by zxeoc** [greydust/mhrise-talisman_organizer#4]
Talisman A: Weakness Exploit 1, Spare Shot 2, slot 3-2
Talisman B: Weakness Exploit 1, Evade Window 1, no slot
After organizing, both of them are locked. But apparently talisman A is better than talisman B.

**Comment by zxeoc** [comment]
By replacing "#decoList <= targetLv" with "#decoList >= targetLv" (in line 21 of organizer.lua), some compare error can be fixed. But there are still some errors:
Talisman A: Evade Window 3, Evade Extender 2, slot 4-1-1
Talisman B: Evade Window 3, Poison Resistance 2, slot 1-1
After organizing, both are locked.

**Comment by ozipone** [comment]
@zxeoc I actually have a similar situation and I think I know what the reason is. I use the setting "keep every level" on Evade Extender and I suspect you might have it too. I had both of these talismans locked.
Talisman A: Peak Performance 2, Evade Extender 2, slot 4-1-1
Talisman B: Peak Performance 1, Evade Extender 1, slot 3-1-1

After swapping to "the more is better" it unlocked talisman B.

But otherwise the change you made seems like it works correctly. I have not yet seen a locked talisman that's clearly inferior to something else or an unlocked one where I couldn't find a better one. I have like 1400 talismans and without the change only 39 were unlocked and with the change that number went to 400.

**Comment by greydust** [comment]
I don't even know why I put a <= at line 21. Perhaps I was sleeping when I wrote this...

**Comment by zxeoc** [comment]
@ozipone  Nope, I use "the more is better" for all skills and have never changed it.

**Comment by greydust** [comment]
I still don't understand why this is happening. Can you set "local debug" to true (first line in debug.lua) and send me the debug log, or send me your save so that I can try it out?

**Comment by zxeoc** [comment]
Sorry, it cannot be done. Since I got over 3000 talismans, which means over 5,000,000 compares. And the debug log concatenates in the memory, so organizing with "local debug" set to true will crash the game. That's why I suggested [add an append method for FS api](https://github.com/praydog/REFramework/issues/797) for REFramework.

**Comment by zxeoc** [comment]
> I still don't understand why this is happening. Can you set "local debug" to true (first line in debug.lua) and send me the debug log, or send me your save so that I can try it out?

Thanks to [Talisman Storage Solution](https://www.nexusmods.com/monsterhunterrise/mods/2413), I can put other talismans in storage file. Now, only two talismans for comparing:
Talisman A: Evade Window 3, Evade Extender 2, slot 4-1-1
Talisman B: Evade Window 3, Poison Resistance 2, slot 1-1

The log file only shows:
```
Checking 1635
Add 1635 to best list
Checking 1636
Add 1636 to best list
```
And here is my save (since there are limitations for attachments, I had to put it in multipart archive files and change the extension names) :
[win64_save.zip.001.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15204582/win64_save.zip.001.zip)
[win64_save.zip.002.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15204583/win64_save.zip.002.zip)
[win64_save.zip.003.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15204584/win64_save.zip.003.zip)
[win64_save.zip.004.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15204585/win64_save.zip.004.zip)
[win64_save.zip.005.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15204586/win64_save.zip.005.zip)
[win64_save.zip.006.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15204588/win64_save.zip.006.zip)
[win64_save.zip.007.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15204589/win64_save.zip.007.zip)

BTW: Did you consider the situation that there may be two or more talismans are exactly the same when comparing?

**Comment by greydust** [comment]
Did you miss a zip part? When I extracted the files it said "Data error : win64_save\SS1_data014Slot.bin" and I cannot load the save file.

**Comment by greydust** [comment]
After re-reading the code, I think I have a guess: The Poison Resistance decorators only have level 1 (1 slot) and level 3(4 slot) options. When the code searches for a level 2 decoration it won't return anything.
```
                if #decoList >= targetLv and decoList[targetLv] and decoList[targetLv] > 0 then
                    for targetDecoLv = decoList[targetLv], 4, 1 do
```

I'll fix the statement.

**Comment by greydust** [comment]
Commit a513d855f2e840043719766349578d577b5dc709 should fix it. @zxeoc Can you try it on your save?

**Comment by zxeoc** [comment]
Sorry for the late reply. I tried again, but the issue still happened.
This is my save:
[win64_save.zip.001.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15251349/win64_save.zip.001.zip)
[win64_save.zip.002.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15251351/win64_save.zip.002.zip)
[win64_save.zip.003.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15251353/win64_save.zip.003.zip)
[win64_save.zip.004.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15251354/win64_save.zip.004.zip)
[win64_save.zip.005.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15251355/win64_save.zip.005.zip)
[win64_save.zip.006.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15251356/win64_save.zip.006.zip)
[win64_save.zip.007.zip](https://github.com/greydust/mhrise-talisman_organizer/files/15251357/win64_save.zip.007.zip)

**Comment by greydust** [comment]
I still could not unzip your file. I tried 7zip, winzip, winrar, and all of them told me that the files are corrupted.

Can you upload your save to a cloud storage (e.g. Google Drive) so that I can download it without unzipping it?

**Comment by zxeoc** [comment]
> I still could not unzip your file. I tried 7zip, winzip, winrar, and all of them told me that the files are corrupted.
> 
> Can you upload your save to a cloud storage (e.g. Google Drive) so that I can download it without unzipping it?

You need to remove the second ".zip" extension file name for every file first. Github doesn't support files like ".001", ".002"... And it's a multipart-archive-zip-file, not 7 different saves.

**Comment by greydust** [comment]
I was able to unzip the file, but still not able to load the save in my game. I'll see if I can reproduce the talisman directly.

**Comment by greydust** [comment]
With the help of [Talisman Storage Solution](https://www.nexusmods.com/monsterhunterrise/mods/2413), I was able to reproduce the case. It turns out that copilot was hallucinating and produced stupid code. It should be fixed with 757118be7bf70440d3b203727438bd9a0fd53599. Can you try again?

**Comment by zxeoc** [comment]
It seems that the problem has been solved. Congrats!
